decko/lib/decko/response.rb

```diff
@@ -16,14 +15,14 @@ def card_redirect url
     end
 
     def deliver format, result, status
-      if format == :file && status == 200
-        send_file(*result)
-      elsif status == 302
+      if status == 302
         card_redirect result
+      elsif format.is_a?(Card::Format::FileFormat) && status == 200
+        send_file(*result)
       else
-        args = { html: result.html_safe, status: status }
-        args[:content_type] = "text/text" if format == :file
-        render args
+        render body: result.html_safe,
+               status: status,
+               content_type: format.content_type
```

Comment:
I'm guessing this is why you used "content_type" rather than "mime_type" as the method, but given how weighted the words "content" and "type" already are for cards, I'm thinking "mime_type" would be less likely to confuse folks (including me)